programs/local/LocalServer.cpp

```diff
@@ -323,12 +323,30 @@ void LocalServer::setupUsers()
     auto & access_control = global_context->getAccessControl();
     access_control.setNoPasswordAllowed(config().getBool("allow_no_password", true));
     access_control.setPlaintextPasswordAllowed(config().getBool("allow_plaintext_password", true));
-    if (config().has("users_config") || config().has("config-file") || fs::exists("config.xml"))
+    if (config().has("config-file") || fs::exists("config.xml"))
     {
-        const auto users_config_path = config().getString("users_config", config().getString("config-file", "config.xml"));
-        ConfigProcessor config_processor(users_config_path);
-        const auto loaded_config = config_processor.loadConfig();
-        users_config = loaded_config.configuration;
+        String config_path = config().getString("config-file","");
+        bool has_user_directories = config().has("user_directories");
+        const auto config_dir = std::filesystem::path{config_path}.remove_filename().string();
```

Comment:
why remove_filename?

Reply:
We want to extract the directory path from the config-file path . Not using remove_filename() will return the config-file path which is something /etc/clickhouse-server/config.xml , However we need only /etc/clickhouse-server.
Please correct me if I am wrong.